### PR TITLE
Switch ACR Helm chart to GHCR for rad init

### LINKED ISSUE #6502: Migrate helm chart to ghcr.io
## Overview of feature request

We had originally planned to move our Helm Chart to `ghcr.io` as part of publicizing the repo. At the last minute we ran into issues pulling and installing the chart in the CLI, so we reverted the changed and kept pulling from ACR.

## Acceptance criteria

Support pull/install in `rad init` from GHCR.

Remove publish to ACR (might want to delay).

Update documentation.

[AB#9841](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/9841)

### PR DESCRIPTION
**Pull request by willdavsmith**
# Description

* Switching ACR to GHCR
* Updating Helm pull config to support OCI registry

## Type of change

- This pull request adds or changes features of Radius and has an approved issue (issue link required).

Fixes: #6502

### Summary
🚚🐳⛵

Support OCI registries for Helm charts and migrate Radius chart to GitHub Container Registry. This change allows the CLI to pull Helm charts from any OCI-compatible registry and updates the default Radius chart URL to use the new hosting service.

> _Sing, O Muse, of the valiant code review_
> _That changed the source of `radiusHelmRepo`_
> _From azure's cloud to github's shining vault_
> _Where OCI charts are stored with skill and fault_

### Walkthrough
*  Support pulling Helm charts from OCI registries ([link](https://github.com/radius-project/radius/pull/6510/files?diff=unified&w=0#diff-35b8f26ef00e63853d6a8303dfdb8755355d7cb6d48ba9449d5a04af62a723cbR32),[link](https://github.com/radius-project/radius/pull/6510/files?diff=unified&w=0#diff-35b8f26ef00e63853d6a8303dfdb8755355d7cb6d48ba9449d5a04af62a723cbL82-R86),[link](https://github.com/radius-project/radius/pull/6510/files?diff=unified&w=0#diff-35b8f26ef00e63853d6a8303dfdb8755355d7cb6d48ba9449d5a04af62a723cbL104-R112))
* Update Radius Helm chart repository URL to GitHub Container Registry in `radiusclient.go` ([link](https://github.com/radius-project/radius/pull/6510/files?diff=unified&w=0#diff-7f93f7a0f0c6c2329155f30e6a4cc6928277acabefab4c28826cb143a23ce90eL36-R36))

### CODE REVIEW

File: pkg/cli/helm/helm.go [hunk 1]
```diff
@@ -79,7 +80,10 @@ func locateChartFile(dirPath string) (string, error) {
 
 func helmChartFromContainerRegistry(version string, config *helm.Configuration, repoUrl string, releaseName string) (*chart.Chart, error) {
 	pull := helm.NewPull()
-	pull.RepoURL = repoUrl
+
+	if !registry.IsOCI(repoUrl) {
+		pull.RepoURL = repoUrl
+	}
```
Comment (rynowak) [inline pkg/cli/helm/helm.go:86]: Can you add some comments here that explain this?

File: pkg/cli/helm/helm.go [hunk 2]
```diff
@@ -101,7 +105,11 @@ func helmChartFromContainerRegistry(version string, config *helm.Configuration,
 
 	pull.DestDir = dir
 
-	_, err = pull.Run(releaseName)
+	chartRef := releaseName
+	if registry.IsOCI(repoUrl) {
+		chartRef = fmt.Sprintf("%s/%s", repoUrl, releaseName)
```
Comment (rynowak) [inline pkg/cli/helm/helm.go:110]: Is this really the **release** name or is this the chart name?

File: pkg/cli/helm/helm.go [hunk 3]
```diff
@@ -101,7 +101,24 @@ func helmChartFromContainerRegistry(version string, config *helm.Configuration,
...
+		pull.RepoURL = repoUrl
+		chartRef = releaseName
+	} else {
+		// For OCI registries (like radius), we will use the
+		// repo URL + the releaseName as the chartRef.
+		// pull.Run("oci://ghcr.io/radius-project/helm-chart/radius")
+		chartRef = fmt.Sprintf("%s/%s", repoUrl, releaseName)
+	}
+
+	_, err = pull.Run(chartRef)
```
Comment (ytimocin) [inline pkg/cli/helm/helm.go:121]: So in the Run function, if the RepoURL is empty (else condition above) then the logic will just go and get the details from the ChartRef?

Comment (willdavsmith) [reply pkg/cli/helm/helm.go:121]: exactly. Setting the RepoURL to an OCI repo did not work when I tried it. setting it here as the chartRef did though.

**Review by ytimocin — APPROVED** [review]
How will we test this in production?

**Comment by willdavsmith** [comment]
> How will we test this in production?

This code path will be called on every rad init and rad install kubernetes :) we will know if it's not working. I'm worried that we don't have any way to test this before merge yet though.

**Comment by rynowak** [comment]
> > How will we test this in production?
> 
> This code path will be called on every rad init and rad install kubernetes :) we will know if it's not working. I'm worried that we don't have any way to test this before merge yet though.

We should make sure that `rad install kubernetes` is covered by our functional tests. That will verify that our dependencies (Helm + GHCR + Publishing to GHCR) are covered. This is also covered by the release verification tests.

The unique logic of `rad init` is pretty well tested by unit tests.

**Comment by willdavsmith** [comment]
> > > How will we test this in production?
> > 
> > 
> > This code path will be called on every rad init and rad install kubernetes :) we will know if it's not working. I'm worried that we don't have any way to test this before merge yet though.
> 
> We should make sure that `rad install kubernetes` is covered by our functional tests. That will verify that our dependencies (Helm + GHCR + Publishing to GHCR) are covered. This is also covered by the release verification tests.
> 
> The unique logic of `rad init` is pretty well tested by unit tests.

Looks like today we test using `rad install kubernetes` but we set the chart path manually (`--chart ./Deploy/chart`) instead of using it from GHCR.

**Comment by ytimocin** [comment]
> > > > How will we test this in production?
> > > 
> > > 
> > > This code path will be called on every rad init and rad install kubernetes :) we will know if it's not working. I'm worried that we don't have any way to test this before merge yet though.
> > 
> > 
> > We should make sure that `rad install kubernetes` is covered by our functional tests. That will verify that our dependencies (Helm + GHCR + Publishing to GHCR) are covered. This is also covered by the release verification tests.
> > The unique logic of `rad init` is pretty well tested by unit tests.
> 
> Looks like today we test using `rad install kubernetes` but we set the chart path manually (`--chart ./Deploy/chart`) instead of using it from GHCR.

Then we should add a GH issue that will add a functional test for the `rad install kubernetes`.

**Pull request merged**